### PR TITLE
Moar nullables

### PR DESCRIPTION
**Pull request by SimonCropp**
needs a squash

### CODE REVIEW

File: src/Serilog/Capturing/PropertyValueConverter.cs [hunk 1]
```diff
@@ -281,7 +283,7 @@ bool TryConvertValueTuple(object value, Destructuring destructuring, Type valueT
             return false;
         }
 
-        bool TryConvertCompilerGeneratedType(object value, Destructuring destructuring, Type valueType, out LogEventPropertyValue result)
+        bool TryConvertCompilerGeneratedType(object value, Destructuring destructuring, Type valueType, [NotNullWhen(true)] out LogEventPropertyValue? result)
```
Comment (nblumhardt) [inline src/Serilog/Capturing/PropertyValueConverter.cs:286]: These are nicer to consume when they're written as:

```csharp
[MaybeNullWhen(false)] out LogEventPropertyValue result
```

I.e. the out var is not nullable, and the compiler checks that the return value has been used. (This is how `Dictionary<K,V>` works when `V` is a non-nullable reference type.)

Comment (SimonCropp) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: but LogEventPropertyValue is a reference type, so it is not comparable to the Dictionary scenario?

Comment (nblumhardt) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: That's a good question; I have a couple of things to check - will tinker and report back :-) 👍

Comment (SimonCropp) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: any outcome from the tinkering?

Comment (nblumhardt) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: Sorry, hectic couple of weeks again! Starting to catch up but slow going :)

Comment (SimonCropp) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: no rush on my side, just making sure it hadnt slipped off your radar

Comment (nblumhardt) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: Thanks :-) :+1:

Comment (nblumhardt) [reply src/Serilog/Capturing/PropertyValueConverter.cs:286]: Thanks for hanging in there, mate - I think your approach is the right one, squashing now!

**Pull request merged**